releasenotes-builder/src/main/java/com/github/checkstyle/Main.java

```diff
@@ -53,36 +54,36 @@ private Main() {
      * @noinspectionreason CallToSystemExit - main method must exit with code
      */
     public static void main(String... args) {
-        int errorCounter;
         List<String> publicationErrors = null;
+        final List<String> errors = new ArrayList<>();
+        boolean hasThrownException = false;
         try {
             final CliProcessor cliProcessor = new CliProcessor(args);
             cliProcessor.process();
             if (cliProcessor.hasErrors()) {
-                printListOf(cliProcessor.getErrorMessages());
-                errorCounter = cliProcessor.getErrorMessages().size();
+                errors.addAll(cliProcessor.getErrorMessages());
             }
             else {
                 final CliOptions cliOptions = cliProcessor.getCliOptions();
                 final Result notesBuilderResult = runGithubNotesBuilder(cliOptions);
+                errors.addAll(notesBuilderResult.getErrorMessages());
                 if (cliOptions.isValidateVersion()) {
                     final List<String> validationErrors = MainProcess
                         .validateNotes(notesBuilderResult.getReleaseNotes(), cliOptions);
-                    notesBuilderResult.addErrors(validationErrors);
+                    errors.addAll(validationErrors);
                 }
-                errorCounter = notesBuilderResult.getErrorMessages().size();
-                if (errorCounter == 0) {
+                if (errors.isEmpty()) {
                     publicationErrors = MainProcess.runPostGenerationAndPublication(
```

Comment:
we should run actual generation of content to show maintainer grouped changes.
if we just print error on version it will be hard to make decision on what to do.
we can use skip publications only if error exists, so we need to split runPostGenerationAndPublication into two methods to make it sure that in case of any problem/error we can generate some files(release notes content) but we do not leak them to public.

Reply:
`runPostGenerationAndPublication` consists of two methods `runPostGeneration` and `runPostPublication`. From what I see `runPostGeneration` does not return errors but rather throw exceptions so we would not even get to running `runPostPublication` if any error is thrown. Maybe just run `runPostGenerationAndPublication` without that `if (errors.isEmpty()) {` ?

Reply:
It might protect execution of this method is any error happen before.
Like if no label is found, how we can do posts generation?

Reply:
I am not sure I understand what you mean exactly.

Reply:
Please make sure that publication doesn't happen if there is any error  happen before.

Reply:
This is going to require a bit of refactoring. I now see the problem with unit tests and why you try to avoid them,

Reply:
Moved logic back to `runPostGenerationAndPublication` and added additional `boolean` parameter to tell if app has encountered any errors before. Personally, not a big fan of that and having a `boolean` parameter but it was the least amount of refactoring. Another solution would be to pass the `error` list to `runPostGenerationAndPublication` so it can check that for itself. Or maybe the best would be to get rid of `runPostGenerationAndPublication` altogether and call `runPostGeneration()` and `runPostPublication()` separately in `Main.main()`. That would require to call them separately int tests as well.

Reply:
If any error happen before publication, we should not publicate

Reply:
>That would require to call them separately int tests as well.

No problems, tests always ok to change.

Reply:
On it.